servant-server/CHANGELOG.md

```diff
@@ -3,6 +3,10 @@
 
 * Add `err422` Unprocessable Entity
   ([#646](https://github.com/haskell-servant/servant/pull/646))
+* Changed `HasServer` instances for `QueryParam` and `QueryParam` to throw 400
+  when parsing fails
```

Comment:
@jkarni I'll edit the changelogs to contain "BREAKING CHANGES" explicitly at the top. But until it, it's just easier to have append only log (also asking others "could you add a changelog entry" is quite vague request")